### PR TITLE
상품 구현

### PR DESCRIPTION
**Pull request by rkdmin**
### 변경사항

**AS-IS**
상품 상세정보 등록 구현
**TO-BE**

### 테스트 
 
- [ ] 테스트 코드
- [ v] API 테스트

### CODE REVIEW

**Comment by rkdmin** [comment]
|File|Coverage [10.23%]|:x:|
|:-|:-:|:-:|
|[ColorType.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Ftype%2FColorType.java)|75.73%|:x:|
|[SizeType.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Ftype%2FSizeType.java)|75.73%|:x:|
|[FitMeApplication.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2FFitMeApplication.java)|37.5%|:x:|
|[ItemController.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fcontroller%2FItemController.java)|27.78%|:x:|
|[CategoryController.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fcontroller%2FCategoryController.java)|20.41%|:x:|
|[MemberController.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fcontroller%2FMemberController.java)|18.84%|:x:|
|[BrandController.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fcontroller%2FBrandController.java)|18.52%|:x:|
|[ModelController.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fcontroller%2FModelController.java)|18.18%|:x:|
|[SellerController.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fcontroller%2FSellerController.java)|17.54%|:x:|
|[ItemService.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fservice%2FItemService.java)|14.86%|:x:|
|[CategoryService.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fservice%2FCategoryService.java)|11.24%|:x:|
|[MemberService.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fservice%2FMemberService.java)|10.33%|:x:|
|[ModelService.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fservice%2FModelService.java)|8.4%|:x:|
|[ItemInfoService.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fservice%2FItemInfoService.java)|8.06%|:x:|
|[SellerService.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fservice%2FSellerService.java)|7.94%|:x:|
|[BrandService.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fservice%2FBrandService.java)|7.87%|:x:|
|[GlobalExceptionHandler.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FGlobalExceptionHandler.java)|4.24%|:x:|
|[Category.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fentity%2FCategory.java)|3.23%|:x:|
|[MemberErrorCode.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2Ftype%2FMemberErrorCode.java)|0%|:x:|
|[ErrorCode.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2Ftype%2FErrorCode.java)|0%|:x:|
|[BrandErrorCode.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2Ftype%2FBrandErrorCode.java)|0%|:x:|
|[ModelErrorCode.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2Ftype%2FModelErrorCode.java)|0%|:x:|
|[CategoryErrorCode.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2Ftype%2FCategoryErrorCode.java)|0%|:x:|
|[ItemInfoErrorCode.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2Ftype%2FItemInfoErrorCode.java)|0%|:x:|
|[SellerErrorCode.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2Ftype%2FSellerErrorCode.java)|0%|:x:|
|[ItemErrorCode.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2Ftype%2FItemErrorCode.java)|0%|:x:|
|[BrandException.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FBrandException.java)|0%|:x:|
|[CategoryException.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FCategoryException.java)|0%|:x:|
|[ItemException.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FItemException.java)|0%|:x:|
|[ModelException.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FModelException.java)|0%|:x:|
|[ErrorResponse.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FErrorResponse.java)|0%|:x:|
|[ItemInfoException.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FItemInfoException.java)|0%|:x:|
|[MemberException.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FMemberException.java)|0%|:x:|
|[SellerException.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fexception%2FSellerException.java)|0%|:x:|
|[RegItem.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fmodel%2FRegItem.java)|0%|:x:|
|[RegItemInfo.java](https://github.com/rkdmin/fitMe/blob/a1ccbbe221e494e4a6233636ccedf9fb48179f51/src%2Fmain%2Fjava%2Fcom%2Fzerobase%2Ffitme%2Fmodel%2FRegItemInfo.java)|0%|:x:|

|Total Project Coverage|7.39%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/zerobase/fitme/service/ItemService.java [hunk 1]
```diff
@@ -33,19 +38,20 @@ public class ItemService {
...
         Model model = modelService.readById(request.getModelId())
-            .orElseThrow(() -> new ItemException(MODEL_NOT_FOUND));
+            .orElseThrow(() -> new ModelException(MODEL_NOT_FOUND));
 
         // 할인 계산
-        request.setSaleRate(calculate(request.getPrice(), request.getSaleRate()));
+        request.setSaleRate(calculateSalePrice(request.getPrice(), request.getSaleRate()));
 
         // 상세정보 저장
-        // 클라이언트에서 받아온 String 데이터를 enum 데이터 리스트로 어떻게 바꾸는지 모르겠습니다.
```
Comment (dongjji) [inline src/main/java/com/zerobase/fitme/service/ItemService.java:46]: 해결하셨는지 모르겠지만 받아온 string 데이터들을 파싱한다음에
enum 클래스의 getValueOf라는 함수를 써서 리스트에 각각 저장해주는 방법은 어떨까요?

Comment (rkdmin) [reply src/main/java/com/zerobase/fitme/service/ItemService.java:46]: 리뷰감사합니다! enum 사용법을 잘 몰라 enum내에 메소드를 사용하는 것을 생각못했는데 감사합니다!!

File: src/main/java/com/zerobase/fitme/controller/CategoryController.java [hunk 2]
```diff
@@ -31,34 +31,35 @@ public class CategoryController {
...
     @PostMapping("/register")
-    public String register(@RequestBody @Valid RegCategory.Request request, BindingResult bindingResult){
+    public ResponseEntity<String> register(@RequestBody @Valid RegCategory.Request request, BindingResult bindingResult){
         // @valid 발생
         validation(bindingResult);
 
         categoryService.register(request);
 
-        return "카테고리 등록이 완료되었습니다.";
+        return ResponseEntity.ok("카테고리 등록이 완료되었습니다.");
```
[the remark this answers is not in the document]
Comment (rkdmin) [reply src/main/java/com/zerobase/fitme/controller/CategoryController.java:40]: 확실히 객체를 리턴하는 방법이 좋은 것 같아요 리팩토링 때 참고하겠습니다 감사합니다!

**Pull request closed without merging**